app/scripts/services/builds.js

```diff
@@ -2,152 +2,85 @@
     };
 
-    BuildsService.prototype.canBuild = function(buildConfig) {
-      if (!buildConfig) {
-        return false;
-      }
-
-      if (buildConfig.metadata.deletionTimestamp) {
-        return false;
-      }
-
-      if (this.isPaused(buildConfig)) {
-        return false;
-      }
-
-      return true;
-    };
-
-    var annotation = $filter('annotation');
-    BuildsService.prototype.usesDeploymentConfigs = function(buildConfig) {
-      var uses = annotation(buildConfig, 'pipeline.alpha.openshift.io/uses');
-      if (!uses) {
-        return [];
-      }
+    return {
+      startBuild: function(buildConfigName, context) {
+        var req = {
+          kind: "BuildRequest",
+          apiVersion: "v1",
+          metadata: {
+            name: buildConfigName
+          }
+        };
+        return DataService.create("buildconfigs/instantiate", buildConfigName, req, context);
+      },
+      cancelBuild: function(build, buildConfigName, context) {
+        var canceledBuild = angular.copy(build);
+        canceledBuild.status.cancelled = true;
+        return DataService.update("builds", canceledBuild.metadata.name, canceledBuild, context);
+      },
+      cloneBuild: function(buildName, context) {
+        var req = {
+          kind: "BuildRequest",
+          apiVersion: "v1",
+          metadata: {
+            name: buildName
+          }
+        };
+        return DataService.create("builds/clone", buildName, req, context);
+      },
+      // Q: should this be in a buildConfig service?
+      isPaused: isPaused,
+      // Q: should this be in a buildConfig service?
+      canBuild: function(buildConfig) {
+        if (!buildConfig) {
+          return false;
+        }
 
-      try {
-        uses = JSON.parse(uses);
-      } catch(e) {
-        Logger.warn('Could not parse "pipeline.alpha.openshift.io/uses" annotation', e);
-        return;
-      }
+        if (buildConfig.metadata.deletionTimestamp) {
+          return false;
+        }
 
-      var depoymentConfigs = [];
-      _.each(uses, function(resource) {
-        if (!resource.name) {
-          return;
+        if (isPaused(buildConfig)) {
+          return false;
         }
 
-        if (resource.namespace && resource.namespace !== _.get(buildConfig, 'metadata.namespace')) {
-          return;
+        return true;
+      },
+      // Q: should this be in a buildConfig service?
+      usesDeploymentConfigs: function(buildConfig) {
+        var uses = annotation(buildConfig, 'pipeline.alpha.openshift.io/uses');
+        if (!uses) {
+          return [];
         }
 
-        if (resource.kind !== 'DeploymentConfig') {
+        if(_.isError(_.attempt(function() {
+          uses = JSON.parse(uses);
+        }))) {
+          Logger.warn('Could not parse "pipeline.alpha.openshift.io/uses" annotation', uses);
           return;
         }
 
-        depoymentConfigs.push(resource.name);
-      });
-
-      return depoymentConfigs;
+        return _.compact(
```

Comment:
No prob, I can pull out this change.  The rest should be exactly the same other than returning the promises so the controllers can handle the `.then()`